types/tool.interfaces.ts

```diff
@@ -50,6 +50,14 @@ export type Tool = {
     readonly type: string;
 };
 
+export enum ArtifactType {
```

Comment:
this can probably be more strongly coupled to https://github.com/compiler-explorer/compiler-explorer/pull/4567/files#diff-0f2c3159b83d36b0cf2000ebf8e22fed9dd53e9afe958f1053af207c4151bb8cR63 - but we can still do that after tool.js has been typescriptified